app/scaffold-controllers/doubleIconsController.js

```diff
@@ -0,0 +1,86 @@
+define([
+    'astro-full',
+    'bluebird',
+    'vendor/backbone-events',
+    'scaffold-plugins/doubleIconsPlugin'
+    ],
+function(Astro, Promise, BackboneEvents, DoubleIconsPlugin) {
+    var DoubleIconsController = function(headerId, generateLeftIcon, generateRightIcon) {
+        this.headerId = headerId;
+        this.generateLeftIcon = generateLeftIcon;
+        this.generateRightIcon = generateRightIcon;
+    };
+
+    var _setLeftIcon = function(doubleIcons, address) {
+        doubleIcons.setLeftIcon(address);
+    };
+
+    var _setRightIcon = function(doubleIcons, address) {
+        doubleIcons.setRightIcon(address);
+    };
+
+    DoubleIconsController.prototype._createDoubleIcons = function(doubleIcons) {
+        this.on('updateLeftIcon', function(param) {
```

Comment:
Do you think you could drop a comment here explaining why we use events here?  It's so that all plugin instances in the nav stack update when we update the icons, right?  It's not obvious if you're coming to this code without the background you have. :smile: